src/modules/trash/trash.usecase.ts

```diff
@@ -0,0 +1,55 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { Folder } from '../folder/folder.domain';
+import { User } from '../user/user.domain';
+import { File } from '../file/file.domain';
+import { FolderUseCases } from '../folder/folder.usecase';
+import { FileUseCases } from '../file/file.usecase';
+
+@Injectable()
+export class TrashUseCases {
+  constructor(
+    private fileUseCases: FileUseCases,
+    private folderUseCases: FolderUseCases,
+  ) {}
+
+  private async deleteFiles(files: Array<File>, user: User) {
+    await Promise.all(
```

Comment:
try to avoid using Promise.all because if there are many files to delete, it may crash the event loop.

Better to delete them sequentially using for/await or to use async.queue

Reply:
Updated